gems/pending/spec/disk/modules/miq_disk_cache_spec.rb

```diff
@@ -0,0 +1,126 @@
+require 'ostruct'
+require 'disk/modules/miq_disk_cache'
+require 'disk/modules/miq_dummy_disk'
+
+describe MiqDiskCache do
+  DUMMY_BLOCKS    = 1000
+  DUMMY_BLOCKSIZE = 4096
+  DUMMY_DATA      = "dummyinfo".freeze
+  before do
+    dummy_info            = OpenStruct.new
+    dummy_info.d_size     = DUMMY_BLOCKS
+    dummy_info.block_size = DUMMY_BLOCKSIZE
+    dummy_info.dummy_data = DUMMY_DATA
+    @dummy_disk           = MiqDummyDisk.new(dummy_info)
+  end
+
+  context ".new" do
+    it "should raise an error, given a bad upstream module" do
+      expect do
+        MiqDiskCache.new(nil)
+      end.to raise_error(RuntimeError)
+    end
+
+    it "should return an MiqDisk object" do
+      miq_cache = MiqDiskCache.new(@dummy_disk)
+      expect(miq_cache).to be_kind_of(MiqDisk)
+    end
+  end
+
+  context "Instance methods" do
+    before do
+      @miq_cache = MiqDiskCache.new(@dummy_disk, 200, 32)
+    end
+
+    describe "#d_size" do
+      it "should return the expected disk cache size" do
+        expect(@miq_cache.d_size).to eq(DUMMY_BLOCKS)
+      end
+    end
+
+    describe "#blockSize" do
+      it "should return the expected disk cache block size" do
+        expect(@miq_cache.blockSize).to eq(DUMMY_BLOCKSIZE)
+      end
+    end
+
+    describe "#size" do
+      it "should return the size of the upstream disk in bytes" do
+        expect(@miq_cache.size).to eq(DUMMY_BLOCKS * DUMMY_BLOCKSIZE)
+      end
+    end
+
+    describe "#lbaStart" do
+      it "should return the expected start logical block address" do
+        expect(@miq_cache.lbaStart).to eq(0)
+      end
+    end
+
+    describe "#lbaEnd" do
+      it "should return the expected end logical block address" do
+        expect(@miq_cache.lbaEnd).to eq(DUMMY_BLOCKS)
+      end
+    end
+
+    describe "#startByteAddr" do
+      it "should return the expected start byte address" do
+        expect(@miq_cache.startByteAddr).to eq(0)
+      end
+    end
+
+    describe "#dInfo" do
+      it "should return the upstream module dInfo data" do
+        expect(@miq_cache.dInfo.dummy_data).to eq(DUMMY_DATA)
+      end
+    end
+
+    describe "#d_write" do
+      it "should return a dummy string" do
```

Comment:
It still needs to adhere to the semantics of the method in question. Otherwise, not only is the test confusing, you're not testing if the proper values are being passed to the upstream module.